### PR TITLE
Proper map code splitting

### PR DESCRIPTION
**Pull request by j8seangel**
Reviewing the map-client project I realized the map was exporting everything (mapbox-gl, pixi.js...) from `index.js` and there were use cases when we don't need it as for example to import the [AVAILABLE_BASEMAPS](https://github.com/GlobalFishingWatch/map-components/compare/map-code-spitting?expand=1#diff-f70efa84e588921f5bed36b357761ffeR3) or `targetMapVessel`. To achieve that it uses https://redux.js.org/recipes/code-splitting#using-replacereducer to make the imports much lighter.

Basically now we can [import the map dynamically](https://github.com/GlobalFishingWatch/map-client/commit/d889ad6c7ec4fa8d3c4f80b95e9e3ecdfc374a8b) which makes the first load paint more than 3 seconds faster.
Before:
image

Now:
image

Also makes the commits hook faster as don't run prettier twice.

### CODE REVIEW

**Comment by nerik** [comment]
Awesome and also thanks for reorganizing the code !
